internal/graph/graph_test.go

```diff
@@ -396,6 +397,44 @@ func TestTopologicallySortWithPriority(t *testing.T) {
 	assert.Error(t, err)
 }
 
+func TestDOTEncoding(t *testing.T) {
+	g := NewGraph[vertex]()
+	adjList := map[string][]string{
+		"v_1": {"v_2", "v_3"},
+		"v_2": {"v_3"},
+		"v_3": {},
+	}
+	expected := `digraph G {
+fontname="Helvetica,Arial,sans-serif"
+node [fontname="Helvetica,Arial,sans-serif"]
+edge [fontname="Helvetica,Arial,sans-serif"]
+n0 [label="v_1"]
+n1 [label="v_2"]
+n2 [label="v_3"]
+n0 -> n1
+n0 -> n2
+n1 -> n2
+}
+`
+	// add vertices
+	for id := range adjList {
+		g.AddVertex(NewV(id))
+	}
+
+	// add edges
+	for id, neighbors := range adjList {
+		for _, neighborId := range neighbors {
+			assert.NoError(t, g.AddEdge(id, neighborId))
+		}
+	}
+
+	// encode to DOT
+	buf := bytes.Buffer{}
+	// sort vertices to ensure deterministic ordering of nodes and edges
+	assert.NoError(t, g.EncodeDOT(&buf, true))
```

Comment:
Move this test into `encode_dot_test.go`.

Also, do you think you could add a test for an empty graph?

Reply:
Yeah for sure let me add a test for that.